### PR TITLE
More e2e test

### PR DESCRIPTION
**Pull request by ElevnLi**

### CODE REVIEW

**Comment by jnothman** [comment]
Strange that cypress thinks Next is covered. Does the test pass on your machine?

**Comment by ElevnLi** [comment]
> Strange that cypress thinks Next is covered. Does the test pass on your machine?

No, as the metadata cannot be applied with the json file uploaded, Next button will stay inactive after clicking "Set Form"

**Comment by ElevnLi** [comment]
The first weedcoco test by checking status to be 'Awaiting Review' is working locally, not sure why it failed here. Might because of timeout not being set correctly?

**Comment by jnothman** [comment]
it says no space left on device: might need to stop the artifact upload until there's a problem.

**Comment by jnothman** [comment]
Curious that layer caching is taking 40 minutes and failing, especially since we've got continue-on-error. :(

The `--build` is necessary, however, if caching is used...

File: search/cypress/fixtures/test_coco/weedcoco.json [hunk 1]
```diff
@@ -0,0 +1,411 @@
+{
```
Comment (jnothman) [inline search/cypress/fixtures/test_coco/weedcoco.json:1]: can we call this coco.json and drop `agcontexts` and `info.metadata` from it?

File: search/cypress/integration/weedai_upload.js [hunk 2]
```diff
@@ -11,11 +11,58 @@ describe('overall upload workflow', () => {
...
+        cy.click_text(/^Submit$/)
+    })
+
+    it('Test Coco Upload', () => {
+        cy.findByText(/^Select annotation format$/).click()
+        cy.findByText("COCO").click()
+        cy.click_text(/^Begin upload$/)
+        cy.get('.dzu-input').attachFile('test_coco/weedcoco.json')
+        cy.click_text(/^Next$/)
+        cy.click_text(/^Apply$/)
```
Comment (jnothman) [inline search/cypress/integration/weedai_upload.js:27]: Does apply need to be clicked when no change is made?

File: search/cypress/support/commands.js [hunk 3]
```diff
@@ -41,12 +41,16 @@ Cypress.Commands.add('register', (username, email, password) => {
...
     cy.get('#username')
     .type(username)
     .should('have.value', username)
     cy.get('#password')
     .type(password)
     .should('have.value', password)
     cy.findAllByRole('button', /^Sign In$/).should('have.length', 1).click()
+})
+
+Cypress.Commands.add('click_text', text => {
```
Comment (jnothman) [inline search/cypress/support/commands.js:54]: can we use camelCase to be consistent with the rest of the library/javascript convention

File: search/cypress/integration/weedai_upload.js (hunk 2 above)
Comment (jnothman) [reply search/cypress/integration/weedai_upload.js:27]: It would be good to test the category conversion too....

**Review by jnothman — COMMENTED** [review]
Otherwise LGTM

**Review by jnothman — APPROVED** [review]

**Pull request merged**